scipy/stats/tests/test_morestats.py

```diff
@@ -2873,105 +2873,107 @@ def test_result(self, correction):
-                                     5.7, 50.4, 357.6, 44.]))
-        incl = -np.deg2rad(np.array([66.1, 68.7, 70.1, 82.1, 79.5,
-                                     73., 69.3, 58.8, 51.4]))
-        data = np.stack((np.cos(incl) * np.cos(decl),
-                         np.cos(incl) * np.sin(decl),
-                         np.sin(incl)),
+        # angles were converted to radians
+        decl = xp.asarray([-5.98996999, -1.08210414, -0.64402649, -0.4712389,
+                           -6.26573201, -0.09948377, -0.87964594, -6.24129741,
+                           -0.76794487])
+        incl = xp.asarray([-1.15366264, -1.1990412 , -1.22347581, -1.43291532,
+                           -1.38753676, -1.27409035, -1.20951317, -1.0262536,
+                           -0.89709924])
+        data = xp.stack((xp.cos(incl) * xp.cos(decl),
+                         xp.cos(incl) * xp.sin(decl),
+                         xp.sin(incl)),
                         axis=1)
 
         dirstats = stats.directional_stats(data)
         directional_mean = dirstats.mean_direction
-        mean_rounded = np.round(directional_mean, 4)
 
-        reference_mean = np.array([0.2984, -0.1346, -0.9449])
-        assert_allclose(mean_rounded, reference_mean)
+        reference_mean = xp.asarray([0.2984, -0.1346, -0.9449])
+        xp_assert_close(directional_mean, reference_mean, atol=1e-4, rtol=1e-4)
 
     @pytest.mark.parametrize('angles, ref', [
         ([-np.pi/2, np.pi/2], 1.),
-        ([0, 2*np.pi], 0.)
+        ([0, 2 * np.pi], 0.)
     ])
-    def test_directional_stats_2d_special_cases(self, angles, ref):
+    def test_directional_stats_2d_special_cases(self, angles, ref, xp):
         if callable(ref):
             ref = ref(angles)
-        data = np.stack([np.cos(angles), np.sin(angles)], axis=1)
+        angles = xp.asarray(angles)
+        data = xp.stack([xp.cos(angles), xp.sin(angles)], axis=1)
         res = 1 - stats.directional_stats(data).mean_resultant_length
-        assert_allclose(res, ref)
+        xp_assert_close(res, ref)
 
-    def test_directional_stats_2d(self):
+    def test_directional_stats_2d(self, xp):
         # Test that for circular data directional_stats
         # yields the same result as circmean/circvar
         rng = np.random.default_rng(0xec9a6899d5a2830e0d1af479dbe1fd0c)
-        testdata = 2 * np.pi * rng.random((1000, ))
-        testdata_vector = np.stack((np.cos(testdata),
-                                    np.sin(testdata)),
+        testdata = xp.asarray(2 * xp.pi * rng.random((1000, )))
+        testdata_vector = xp.stack((xp.cos(testdata),
+                                    xp.sin(testdata)),
                                    axis=1)
         dirstats = stats.directional_stats(testdata_vector)
         directional_mean = dirstats.mean_direction
         directional_mean_angle = np.arctan2(directional_mean[1],
                                             directional_mean[0])
```

Comment:
Re:atan2: When there is a feature you need in a test that is not in the regular library but has been in the standard long enough for `array_api_compat` to catch up, we have typically used the `array_api_compat` version of the library. (For example, search for `xp_test`.)

```suggestion
        xp_test = array_namespace(directional_mean)  # np doesn't have atan2
        directional_mean_angle = np.arctan2(directional_mean[1],
                                            directional_mean[0])
```